test/hotspot/jtreg/compiler/c2/irTests/TestFloat16ScalarOperations.java

```diff
@@ -287,81 +290,148 @@ public void testDivByPOT() {
         short res = 0;
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() + INEXACT_FP16);
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() - INEXACT_FP16);
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() * INEXACT_FP16);
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() / INEXACT_FP16);
-         assertResult(Float.float16ToFloat(res), 32.125f, "testInexactFP16ConstantPatterns");
+        res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() + INEXACT_FP16);
+        res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() - INEXACT_FP16);
+        res += Float.floatToFloat16(INEXACT_FP16 * POSITIVE_ZERO_VAR.floatValue());
+        res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() / INEXACT_FP16);
+        return res;
+    }
+
+    @Check(test="testInexactFP16ConstantPatterns")
+    public void checkInexactFP16ConstantPatterns(short actual) throws Exception {
+        TestFramework.deoptimize(TestFloat16ScalarOperations.class.getMethod("testInexactFP16ConstantPatterns"));
+        short expected = testInexactFP16ConstantPatterns();
+        Verify.checkEQ(expected, actual);
     }
 
     @Test
-    @IR(counts = {IRNode.ADD_HF, " 0 ", IRNode.SUB_HF, " 0 ", IRNode.MUL_HF, " 0 ", IRNode.DIV_HF, " 0 "},
+    @IR(counts = {IRNode.ADD_HF, " >0 ", IRNode.SUB_HF, " >0 ", IRNode.MUL_HF, " >0 ", IRNode.DIV_HF, " >0 "},
         applyIfCPUFeatureOr = {"avx512_fp16", "true", "zfh", "true"})
-    @IR(counts = {IRNode.ADD_HF, " 0 ", IRNode.SUB_HF, " 0 ", IRNode.MUL_HF, " 0 ", IRNode.DIV_HF, " 0 "},
+    @IR(counts = {IRNode.ADD_HF, " >0 ", IRNode.SUB_HF, " >0 ", IRNode.MUL_HF, " >0 ", IRNode.DIV_HF, " >0 "},
         applyIfCPUFeatureAnd = {"fphp", "true", "asimdhp", "true"})
     @Warmup(10000)
-    public void testSNaNFP16ConstantPatterns() {
+    public short testSNaNFP16ConstantPatterns() {
         short res = 0;
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() + SNAN_FP16);
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() - SNAN_FP16);
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() * SNAN_FP16);
-         res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() / SNAN_FP16);
-         assertResult(Float.float16ToFloat(res), -32768.0f, "testSNaNFP16ConstantPatterns");
+        res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() + SNAN_FP16);
+        res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() - SNAN_FP16);
+        res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() * SNAN_FP16);
+        res += Float.floatToFloat16(POSITIVE_ZERO_VAR.floatValue() / SNAN_FP16);
+        return res;
+    }
+
+    @Check(test="testSNaNFP16ConstantPatterns")
+    public void checkSNaNFP16ConstantPatterns(short actual) throws Exception {
+        TestFramework.deoptimize(TestFloat16ScalarOperations.class.getMethod("testSNaNFP16ConstantPatterns"));
```

Comment:
Oh wow, I have never seen this pattern used. Cool idea! Do you know what impact this has on test runtime?

Reply:
IIUC, since entier framework is based on whitebox APIs hence @Check annotated method is only invoked once after each @Test annotated method execution, I don't see much impact on test execution time here.